packages/gatsby-transformer-sharp/src/fragments.js

```diff
@@ -119,7 +119,7 @@ export const GatsbyImageSharpFluid = graphql`
  * Presentation sizes to make sure a fluid container does not overflow
  * @type {Fragment}
  */
-export const GatsbyImageSharpFluidLimitPresentationSize = `
+export const GatsbyImageSharpFluidLimitPresentationSize = graphql`
   fragment GatsbyImageSharpFluidLimitPresentationSize on ImageSharpFluid {
     maxHeight: presentationHeight
```

Comment:
@wardpeet if you are setting the maxHeight/maxWidth to the presentationHeight/presentationWidth I'm not sure this will end up being correct, see this comment of mine on the issue of using presentationHeight/Width as maxHeight/Width. [comment](https://github.com/gatsbyjs/gatsby/pull/17006#issuecomment-532056611).  Basically the presentationHeight/Width max out at the images dimensions, so if your image is 800x800 and you set maxWidth to 1500, it will still only be maxWidth 800.

Reply:
and this [comment](https://github.com/gatsbyjs/gatsby/pull/17006#issuecomment-532757200) also.

Reply:
I was finally able to get the fragments to update and working using: 

```
       childImageSharp {
          fluid(maxWidth: 500) {
            ...GatsbyImageSharpFluid
            ...GatsbyImageSharpFluidLimitPresentationSize
          }
        }
```

and it does appear to be using presentationHeight/Width.  So any maxHeight/width will max out at the image dimensions.  This is why I had the logic in `gatsby-plugin-sharp` 

```
  let maxWidth
  let maxHeight
  if (options.maxHeight || options.maxWidth) {
    maxWidth = options.maxWidth
      ? `${options.maxWidth}px`
      : `${options.aspectRatio * options.maxHeight}px`
    maxHeight = options.maxHeight
      ? `${options.maxHeight}px`
      : `${options.aspectRatio * options.maxWidth}px`
  }
```
and specifically returned `maxHeight/maxWidth` it as part of the fluid return object.

Reply:
Setting it to presentationWidth/height feels more correct as your the container shouldn't be bigger than image. If you need that, you'll need a bigger image, else you'll end up with quality drop.

If you want to center the image or override the behavior you can still add a wrapper or set the maxWidth/maxHeight yourself on gatsby-image.

Reply:
If this makes more sense I'm good with it.